internal/pkg/api/server/templates/service/mapper.go

```diff
@@ -189,6 +189,9 @@ func InputsResponse(stepsGroups input.StepsGroupsExt) (out *Inputs) {
 					Default:     in.DefaultOrEmpty(),
 					Options:     OptionsResponse(in.Options),
 				}
+				if in.ComponentId != "" {
+					inputResponse.ComponentID = &in.ComponentId
+				}
```

Comment:
Nemůže se stát, že by `in.ComponentId` nebylo vůbec definovaný? Potom by tahle podmínka nefungovala ne?

Reply:
Jé promiň, to jsem ti měl napsat. Nemůže, Go je typově striktní. V `in` je struktura typu `Input` a ta má definovanou property `ComponentId` a tak ji každá proměnná toho typu musí vrátit. A pokud se do tý property nic neuložilo, tak vrátí nulovou hodnotu pro ten daný typ. 

Jo a `ComponentId` je `string` a nulová hodnota stringů je prázdný řetězec. Proto je `!= ""` v tom ifu, nejde tam napsat nic jako `!= nil` ani `if !in.ComponentId`.

(Super že se ptáš, aspoň minimalizujeme riziko chyb. 🙂)

![CleanShot 2022-06-24 at 08 57 36@2x](https://user-images.githubusercontent.com/215660/175480657-363aea87-32a6-4a84-98e8-f7268677ec49.jpg)

Reply:
Díky 🙂